server/src/test/java/org/elasticsearch/action/bulk/TransportBulkActionTests.java

```diff
@@ -303,15 +306,23 @@ public void testOnlySystem() {
         );
         List<String> onlySystem = List.of(".foo", ".bar");
         assertTrue(TransportBulkAction.isOnlySystem(buildBulkRequest(onlySystem), indicesLookup, systemIndices));
+        /* Test forwarded bulk requests (that are serialized then deserialized) */
+        assertTrue(TransportBulkAction.isOnlySystem(buildBulkStreamRequest(onlySystem), indicesLookup, systemIndices));
```

Comment:
The original problem is that system thread pool is used unexpectedly because `isOnlySystem` always return `true` when the request is forwarded. So I think we should have a test that demonstrates `isOnlySystem` returns `false` when the indices are not all system indices.

Reply:
Hi Yang, I also added assertFalse for nonSystem and mixed index sets below. Does that cover your concern or were you thinking of something else ?

Reply:
Right, they look good to me. Somehow missed them initially 🤦